### ISSUE TITLE
Error when updating to Beta5 NuGet packages: No service for type 'Microsoft.Framework.Runtime.ILibraryManager' has been registered.

### ISSUE THREAD

**Issue by nh43de** [aspnet/Mvc#2576]
Got this error when trying to update packages to latest version for testing in a different project. Was able to reproduce using the **Web API** template using the steps below

1. Create a new vNext Web API Project in VS2015RC
2. Update packages in project.json to use latest package, like this

  "dependencies": {
    "Microsoft.AspNet.Mvc": "6.0.0-\*",
    "Microsoft.AspNet.Server.IIS": "1.0.0-\*",
    "Microsoft.AspNet.Server.WebListener": "1.0.0-\*",
    "Microsoft.AspNet.StaticFiles": "1.0.0-\*"
  }

3. Update dnx to version 1.0.0-beta5-11781, restart visual studio

4. Restore packages, build, and run. Produces this error message:

No service for type 'Microsoft.Framework.Runtime.ILibraryManager' has been registered.

**Comment by Eilon** [comment]
You want to use `6.0.0-beta4`, `1.0.0-beta4`, etc. packages. If you use just `6.0.0` and have the ASP.NET MyGet feeds configured, you'll get beta5 packages, yet you're using a DNX from beta4. All the packages you use, plus the DNX you use, must all be from the same "ship train."

**Comment by Eilon** [comment]
For reference, this is the dependencies section of a brand new ASP.NET 5 project in VS2015 RC:

```json
  "dependencies": {
    "EntityFramework.SqlServer": "7.0.0-beta4",
    "EntityFramework.Commands": "7.0.0-beta4",
    "Microsoft.AspNet.Mvc": "6.0.0-beta4",
    "Microsoft.AspNet.Mvc.TagHelpers": "6.0.0-beta4",
    "Microsoft.AspNet.Authentication.Cookies": "1.0.0-beta4",
    "Microsoft.AspNet.Authentication.Facebook": "1.0.0-beta4",
    "Microsoft.AspNet.Authentication.Google": "1.0.0-beta4",
    "Microsoft.AspNet.Authentication.MicrosoftAccount": "1.0.0-beta4",
    "Microsoft.AspNet.Authentication.Twitter": "1.0.0-beta4",
    "Microsoft.AspNet.Diagnostics": "1.0.0-beta4",
    "Microsoft.AspNet.Diagnostics.Entity": "7.0.0-beta4",
    "Microsoft.AspNet.Identity.EntityFramework": "3.0.0-beta4",
    "Microsoft.AspNet.Server.IIS": "1.0.0-beta4",
    "Microsoft.AspNet.Server.WebListener": "1.0.0-beta4",
    "Microsoft.AspNet.StaticFiles": "1.0.0-beta4",
    "Microsoft.AspNet.Tooling.Razor": "1.0.0-beta4",
    "Microsoft.Framework.ConfigurationModel.Json": "1.0.0-beta4",
    "Microsoft.Framework.ConfigurationModel.UserSecrets": "1.0.0-beta4",
    "Microsoft.Framework.CodeGenerators.Mvc": "1.0.0-beta4",
    "Microsoft.Framework.Logging": "1.0.0-beta4",
    "Microsoft.Framework.Logging.Console": "1.0.0-beta4",
    "Microsoft.VisualStudio.Web.BrowserLink.Loader": "14.0.0-beta4"
  },
```

**Comment by nh43de** [comment]
Thanks for the help :) Forgot to mention I did update dnx to latest version (v1.0.0-beta5-11781).

It looks like when the packages restore after changing to x.x.x.x-\*, it looks like they all have different build numbers. I'm assuming this is where the incompatibility arises?

**Comment by Eilon** [comment]
The specific build numbers will vary a lot, because each GitHub repo ends up having its own build numbers.

Please note that when running from VS, I believe VS will choose whichever DNX you have aliased as "default".

```
Active Version           Runtime Architecture Location                       Alias
------ -------           ------- ------------ --------                       -----
       1.0.0-beta4       clr     x64          C:\Users\elipton\.dnx\runtimes
       1.0.0-beta4       clr     x86          C:\Users\elipton\.dnx\runtimes default
       1.0.0-beta4       coreclr x64          C:\Users\elipton\.dnx\runtimes
       1.0.0-beta4       coreclr x86          C:\Users\elipton\.dnx\runtimes
  *    1.0.0-beta5-11781 clr     x86          C:\Users\elipton\.dnx\runtimes
       1.0.0-beta5-11781 coreclr x86          C:\Users\elipton\.dnx\runtimes
```

That's from my system - I have 1.0.0-beta4 as "default", but my current active DNX is a beta5 build.

You can change the "default" alias to point to a different build: `dnvm alias default 1.0.0-xyz123` and then restart VS.

Or, in VS, right-click on the project, choose Properties, and then pick a different DNX version/flavor/bitness to use to run the app.

**Comment by nh43de** [comment]
New packages, new errors today. Same project, restored packages to latest beta5's, and got this error.

System.MissingMethodException
Method not found: 'Microsoft.Framework.ConfigurationModel.IConfigurationSourceRoot Microsoft.Framework.ConfigurationModel.ConfigurationExtensions.AddEnvironmentVariables(Microsoft.Framework.ConfigurationModel.IConfigurationSourceRoot)'.

DNX is latest github (dev) build as of today, 20 May 2015 10:45 AM CDT

**Comment by kirthik** [comment]
These errors are happening because of fixes made to issues [#195](https://github.com/aspnet/Configuration/issues/195), [#196](https://github.com/aspnet/Configuration/issues/196), [#198](https://github.com/aspnet/Configuration/issues/198). Please change the name of `IConfigurationSourceRoot` to `IConfigurationBuilder` and change assembly `Microsoft.Framework.ConfigurationModel` to `Microsoft.Framework.Configuration`.Also add `Microsoft.Framework.Configuration.EnvironmentVariables` assembly to `project.json`

**Comment by danroth27** [comment]
Looks like this was due to recent API renames.
